src/overlays/actors/ovl_Bg_Ikana_Dharma/z_bg_ikana_dharma.c

```diff
@@ -28,47 +34,221 @@ const ActorInit Bg_Ikana_Dharma_InitVars = {
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_ikana_obj_Colheader_000C50);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    if (BGIKANADHARMA_IS_CHILD(&this->dyna.actor) == 0) {
+        f32 segmentY = this->dyna.actor.world.pos.y;
+        s32 i = 0;
+        s32 numSegments = BGIKANADHARMA_NUM_SEGMENTS(&this->dyna.actor);
+        for (; i < numSegments; i++) {
+            segmentY += 60.0f;
+            Actor_SpawnAsChildAndCutscene(&globalCtx2->actorCtx, globalCtx, ACTOR_BG_IKANA_DHARMA,
+                                          this->dyna.actor.world.pos.x, segmentY, this->dyna.actor.world.pos.z,
+                                          this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y,
+                                          this->dyna.actor.shape.rot.z, BGIKANADHARMA_FLAG_IS_CHILD,
+                                          this->dyna.actor.cutscene, this->dyna.actor.unk20, NULL);
+        }
+        this->dyna.actor.bgCheckFlags |= 1;
+    }
+    BgIkanaDharma_BeginNormalState(this);
+}
+
+void BgIkanaDharma_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgIkanaDharma* this = THIS;
+
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    if (sFirstHitBgIkanaDharma == this) {
+        sFirstHitBgIkanaDharma = NULL;
+    }
+}
+
+void BgIkanaDharma_BeginNormalState(BgIkanaDharma* this) {
+    this->actionFunc = BgIkanaDharma_UpdateNormalState;
+    this->dyna.actor.speedXZ = 0.0f;
+}
+
+void BgIkanaDharma_UpdateNormalState(BgIkanaDharma* this, GlobalContext* globalCtx) {
+    s32 wasHit = (this->collider.base.acFlags & AC_HIT) != 0;
+    Player* player = GET_PLAYER(globalCtx);
+    BgIkanaDharma* this2 = this;
+    s32 temp_v0_3;
+
+    if (wasHit) {
+        this->collider.base.acFlags &= ~AC_HIT;
+    }
+    if (wasHit && sFirstHitBgIkanaDharma == NULL) {
+        s32 temp_v0_2;
+        sFirstHitBgIkanaDharma = this2;
+        Flags_SetSwitch(globalCtx, BGIKANADHARMA_GET_SWITCHFLAG(&this->dyna.actor));
+        temp_v0_3 = (s16)(this->dyna.actor.yawTowardsPlayer + 0x8000);
+        temp_v0_2 = ((s16)(player->actor.shape.rot.y - temp_v0_3) >> 1);
+        this->dyna.actor.world.rot.y = temp_v0_3 + temp_v0_2 + 0xF000;
+        this->dyna.actor.speedXZ = 20.0f;
+        Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_DARUMA_VANISH);
+        BgIkanaDharma_BeginWaitToStartCutscene(this);
+    } else if ((this->dyna.actor.flags & 0x40) == 0x40 && sFirstHitBgIkanaDharma == NULL &&
+               this->dyna.actor.xzDistToPlayer < 420.0f) {
+        temp_v0_3 = (s16)(this->dyna.actor.yawTowardsPlayer - player->actor.shape.rot.y);
+        if (temp_v0_3 < 0) {
+            temp_v0_3 = -temp_v0_3; // this line is the reason temp_v0_3 isn't s16
+        }
+        if (temp_v0_3 >= 0x4001) {
```

Comment:
I think this might be
```suggestion
        if (ABS((s16)(this->dyna.actor.yawTowardsPlayer - player->actor.shape.rot.y)) > 0x4000) {
```
If not, I suspect it's still an `ABS`, and please use the rounder number in the comparison if possible.

Reply:
When ABS is inline like that, the compiler wants to use a different register for input and after. Not sure how to solve that except by assigning it back to the same variable. 0x4000 does work though.